.github/workflows/check-markdown-links.yml

```diff
@@ -19,7 +19,7 @@ jobs:
           persist-credentials: false
 
       - name: Check markdown links
-        uses: gaurav-nelson/github-action-markdown-link-check@5c5dfc0ac2e225883c0e5f03a85311ec2830d368
```

Comment:
It looks like this upgrade requires some additional fixes on our side. Let me know if you'd like to make these changes, otherwise I can go ahead and take care of this version upgrade and let you know when it's ready :)

Reply:
Right, this update fixed errors due to old version of "markdown-link-check" plugin and showed bad links in "definishions.md" file, which I just fixed.

And now only 2 bad links left in `docker.md`.
But it's not related to this change, I see it already happened in another repository:
https://github.com/dotnet/dotnet-docker/issues/5097

So I'll leave it to you guys.

Reply:
Sounds good, let's go ahead and just ignore `hub.docker.com` links from validation. 

We just need to add another pattern to this configuration file:
https://github.com/dotnet/dotnet-monitor/blob/6a13c8f695673e0e63ccce3266deeb9b4e4e9506/.github/linters/check-markdown-links-config.json#L2-L15


Something like:
```json
        {
            "pattern": "^https://hub\\.docker.com/.*"
        }
```

Let me know if you'd prefer we make this change, otherwise feel free to update this PR with it!

Reply:
Done, just updated PR